modules/features/search/src/main/java/au/com/shiftyjelly/pocketcasts/search/SearchHandler.kt

```diff
@@ -120,28 +120,40 @@ class SearchHandler @Inject constructor(
             } else {
                 analyticsTracker.track(AnalyticsEvent.SEARCH_PERFORMED, AnalyticsProp.sourceMap(source))
                 loadingObservable.accept(true)
+
+                var globalSearch = GlobalServerSearch(searchTerm = it)
                 val podcastServerSearch = serverManager
                     .searchForPodcastsRx(it)
+                    .map { podcastSearch ->
+                        globalSearch = globalSearch.copy(podcastSearch = podcastSearch)
+                        globalSearch
+                    }
                     .subscribeOn(Schedulers.io())
                     .toObservable()
 
-                val episodesServerSearch = cacheServerManager
-                    .searchEpisodes(it)
+                if (BuildConfig.SEARCH_IMPROVEMENTS_ENABLED) {
+                    val episodesServerSearch = cacheServerManager
+                        .searchEpisodes(it)
+                        .map { episodeSearch ->
+                            globalSearch = globalSearch.copy(episodeSearch = episodeSearch)
+                            globalSearch
+                        }
+                        .subscribeOn(Schedulers.io())
+                        .toObservable()
```

Comment:
It doesn't really do any harm, but I don't think we actually need to convert this `Single` to an `Observable` because there is an `Observable::mergeWith` method that accepts a `Single` as its parameter. Certainly not a blocker (and maybe I'm overlooking something), so feel free to leave it as-is.

Reply:
You're right, I verified it was not needed so removed it in 13c6d2f.